client/my-sites/checkout/checkout-system-decider.js

```diff
@@ -216,10 +219,20 @@ function getCheckoutVariant(
 		'concierge-session',
 		'ecommerce',
 		'ecommerce-2-years',
+		'jetpack_backup_daily',
```

Comment:
Just to be extra-safe, instead of directly adding these jetpack products to the array of allowed slugs, you could add them to the array only if `config( 'env_id' ) === 'production'`.